src/components/slider/slider.js

```diff
@@ -5,7 +5,70 @@
   angular.module('material.components.slider', [
     'material.core'
   ])
-  .directive('mdSlider', SliderDirective);
+  .directive('mdSlider', SliderDirective)
+  .directive('mdSliderContainer', SliderContainerDirective);
+
+/**
+ * @ngdoc directive
+ * @name mdSliderContainer
+ * @module material.components.slider
+ * @restrict E
+ * @description
+ * The `<md-slider-container>` contains slider with two other elements.
+ *
+ *
+ * @usage
+ * <h4>Normal Mode</h4>
+ * <hljs lang="html">
+ * </hljs>
+ */
+function SliderContainerDirective() {
+  return {
+    compile: function (elem) {
+      var slider = angular.element(elem[0].getElementsByTagName('md-slider'));
```

Comment:
I believe you can replace this with

    var slider = elem.find('md-slider');

since you are finding an element, and not using a CSS query selector.